src/components/BoiteMusicApp.vue

```diff
@@ -0,0 +1,262 @@
+<template>
+  <div class="boitemusic">
+    <h1>AshleyChafaeMusic</h1>
+
+    <!-- Section de la piste en cours de lecture -->
+    <div class="current-track">
+      <p>Now playing: {{ currentPlaylistName }}</p> <!-- Afficher le nom de la playlist -->
+    </div>
+
+    <!-- Image statique -->
+    <div class="album-art" :class="{ playing: isPlaying }">
+      <img :src="currentTrack ? currentTrack.art : albumImage" alt="Album Art" />
+    </div>
+
+    <!-- Contrôles de lecture -->
+    <div class="controls">
+      <button @click="togglePlay">{{ isPlaying ? 'Pause' : 'Play' }}</button>
+      <button @click="skipTrack">Next</button>
+    </div>
+
+    <!-- Barre de progression interactive -->
+    <div class="progress-bar" @click="seekAudio">
+      <div class="progress" :style="{ width: progress + '%' }"></div>
+    </div>
+
+    <!-- Élément audio -->
+    <audio ref="audio" :src="currentTrack ? currentTrack.url : ''" @ended="handleEnd"></audio>
+
+    <!-- Options de lecture (mode de répétition) -->
+    <div class="playback-options">
+      <label>Playback mode:</label>
+      <input type="radio" id="repeatList" value="repeatList" v-model="playbackMode" />
+      <label for="repeatList">Repeat list</label>
+      <input type="radio" id="repeatTrack" value="repeatTrack" v-model="playbackMode" />
+      <label for="repeatTrack">Repeat track</label>
+      <input type="radio" id="noRepeat" value="noRepeat" v-model="playbackMode" />
+      <label for="noRepeat">Don't repeat</label>
+    </div>
+
+    <!-- Sélection de fichier MP3 local ou URL -->
+    <div>
+      <label for="fileInput">Select MP3 file:</label>
+      <input type="file" id="fileInput" accept="audio/mp3" @change="handleFileInput" />
+    </div>
+
+    <div>
+      <label for="urlInput">Or enter MP3 URL:</label>
+      <input type="text" id="urlInput" v-model="urlInput" placeholder="Enter MP3 URL" @keyup.enter="handleUrlInput" />
+      <button type="button" @click="handleUrlInput">Add MP3 URL</button>
+    </div>
+
+    <!-- Liste de lecture -->
+    <MusicPlaylist 
+      :playlist="currentPlaylist" 
+      :currentTrack="currentTrack" 
+      @playTrack="playTrack" 
+      @deleteTrack="deleteTrack" 
+    />
+    <PlaylistManager 
+      v-if="showPlaylistManager" 
+      @selectPlaylist="handleSelectPlaylist" 
+      @close="showPlaylistManager = false" 
+    />
+
+    <!-- Bouton pour aller à la page de gestion des playlists -->
+    <button @click="goToPlaylist">Gérer les playlists</button>
+  </div>
+</template>
+
+<script>
+import MusicPlaylist from './MusicPlaylist.vue';
+import albumImage from '@/assets/album.jpg';
+import PlaylistManager from './PlaylistManager.vue';
+
+export default {
+  name: "BoiteMusicApp",
+  components: {
+    MusicPlaylist,
+    PlaylistManager,
+  },
+  data() {
+    return {
+      currentTrack: null,
+      playbackMode: 'noRepeat',
+      playlists: [], // Liste de toutes les playlists
```

Comment:
Votre code n'était pas fonctionnel !
Il a fallu changer cette ligne pour avoir la moindre de fonctionnalité. Veillez à tester l'application avant la soumission finale.
```suggestion
      playlists: [{tracks: []}],
```